File: src/ca/qc/bdeb/inf203/model/combatants/AtomicRose.java
package ca.qc.bdeb.inf203.model.combatants;

import ca.qc.bdeb.inf203.model.*;
import java.util.HashMap;

/**
 * Rose nucléaire qui tue tout sur le terrain.
 *
 * @author Guillaume Riou, Nicolas Hurtubise
 */
public class AtomicRose extends Combattant {

    public AtomicRose() {
        super();
        initialise();
    }

    @Override
    protected final void initialise() {
        super.initialise();
        this.vie = 30;
        this.gentil = true;
        this.hitbox.width = 80;
        this.hitbox.height = 80;
        this.lineOfSight.x = 0;
        this.lineOfSight.y = 0;
        this.lineOfSight.width = Terrain.TAILLE_CASE_X * Terrain.CASES_X;
        this.lineOfSight.height = Terrain.TAILLE_CASE_Y * Terrain.CASES_Y;

        this.animationFrameRate = 6;

        this.nbrImagesAnimation = new HashMap<>();
        this.nbrImagesAnimation.put(Etat.ATTENTE, 1);
        this.nbrImagesAnimation.put(Etat.ATTAQUE, 11);
        this.etat = Etat.ATTENTE;

        this.vitesseAction.put(Action.ATTAQUE, 1 / 3f);
        this.vitesseAction.put(Action.ACTION, 1 / 3f);
        

        this.sprites.put(Etat.ATTENTE, new RepresentationImage(new String[]{"plants", "atomic-rose"}));
        this.sprites.put(Etat.ATTAQUE, new RepresentationImage(new String[]{"explosion"}));
    }

    @Override
    public Entite tic() {
        Entite entite = null;
        switch (etat) {
            case ATTAQUE:
                entite = action(getNbrActions(Action.ATTAQUE));
                break;
        }

        return entite;
    }

    /**
     * 
     * @param nbFois si c'est 0, pas d'explosion, sinon, explose.
     * @return une explosion.
     */
    @Override
    public Entite action(int nbFois) {

        if (nbFois > 0) {
            return new ExplosionNucleaire();
        }

        return null;
    }
}

File: assets/js/model/combatants/atomic-rose.js
/**
 * Rose nucléaire qui tue tout sur le terrain.
 *
 * @author Guillaume Riou, Nicolas Hurtubise
 */
function AtomicRose() {
    Combattant.apply(this);
    this.combattant_init();
    this.init();
}
AtomicRose.prototype = new Combattant();

AtomicRose.prototype.init = function() {
    this.vie = 30;
    this.gentil = true;
    this.hitbox.w = 80;
    this.hitbox.h = 80;
    this.lineOfSight.x = 0;
    this.lineOfSight.y = 0;
    this.lineOfSight.w = Terrain.TAILLE_CASE_X * Terrain.CASES_X;
    this.lineOfSight.h = Terrain.TAILLE_CASE_Y * Terrain.CASES_Y;

    this.animationFrameRate = 6;

    this.nbrImagesAnimation = {};
    this.nbrImagesAnimation[navigator.Etat.ATTENTE] = 1;
    this.nbrImagesAnimation[navigator.Etat.ATTAQUE] = 11;
    this.etat = navigator.Etat.ATTENTE;

    this.vitesseAction[navigator.Action.ATTAQUE] = 1/3;
    this.vitesseAction[navigator.Action.ACTION] = 1/3;

    this.sprites[navigator.Etat.ATTENTE] = new RepresentationImage(["plants", "atomic-rose"]);
    this.sprites[navigator.Etat.ATTAQUE] = new RepresentationImage(["explosion"]);
};

AtomicRose.prototype.tic = function() {
    var entite = null;

    switch (this.etat) {
        case navigator.Etat.ATTAQUE:
            entite = this.action(this.getNbrActions(navigator.Action.ATTAQUE));
            break;
    }

    return entite;
};

/**
 * @param nbFois si c'est 0, pas d'explosion, sinon, explose.
 * @return Entite une explosion.
 */
AtomicRose.prototype.action = function(times) {

    if (times > 0) {
        return new ExplosionNucleaire();
    }

    return null;
};
